### PR TITLE
[WIP] Migrate to Circle CI 2.0

### PR DESCRIPTION
**Pull request by camillevilla**
[Work in progress]

Circle CI 1.0 will no longer be supported after August 31, 2018.

To-do:
- [x] Update test build configs
  - [x] Resolve yarn - webpacker issue (comment below)
  - [x] Verify whether issue above is caused by choosing the wrong Docker image? 😅
- [x] Update deployment settings
- [x] Clean up comments from autogen file

Resources:
- [Circle CI docs: sample configs](https://circleci.com/docs/2.0/sample-config/)
- [Circle CI docs: Migrating from 1.0 to 2.0](https://circleci.com/docs/2.0/migrating-from-1-2/#steps-to-configure-required-20-keys)
- [Circle CI forum: Yarn build failure - old version finding incompatible packages](https://discuss.circleci.com/t/yarn-build-failure-old-version-finding-incompatible-packages/12259)
- [Circle CI docs: Using Yarn in Circle CI](https://circleci.com/docs/2.0/yarn/)

### CODE REVIEW

**Comment by camillevilla** [comment]
😢
```
#!/bin/bash --login
yarn
yarn install v0.21.3
[1/4] Resolving packages...
success Nothing to install.
success Saved lockfile.
$ cd client && yarn install
yarn install v0.21.3
[1/4] Resolving packages...
[2/4] Fetching packages...
error @rails/webpacker@3.4.3: The engine "yarn" is incompatible with this module. Expected version ">=0.25.2".
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Exited with code 1
```

**Comment by baohouse** [comment]
I guess we need to update yarn version?

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        command: bundle exec rake db:create db:schema:load --trace
+        environment:
+          RAILS_ENV: test
+          RACK_ENV: test
+    # Test
+    #   This would typically be a build job when using workflows, possibly combined with build
+    - run: RAILS_ENV=test bundle exec rake db:create db:schema:load
+    - run: cd client && RAILS_ENV=test bundle exec rake react_on_rails:locale && yarn run lint:setup && yarn run lint && yarn run build:test
+    - run: bundle exec rspec --format progress --format RspecJunitFormatter -o $CIRCLE_TEST_REPORTS/rspec.xml
+    - run: RAILS_ENV=test bundle exec rake jasmine:ci
```
Comment (julianguyen) [inline .circleci/config.yml]: We should also run `cd client && yarn test` after this

**Comment by julianguyen** [comment]
Hey, @camillevilla do you mind if I try to finish up your WIP branch? I know you've been busy with various things. Sending you positive vibes 💜

**Comment by camillevilla** [comment]
Thank you for finishing this, @julianguyen! Apologies for letting this get away from me. 💜

**Pull request merged**